packages/components/Notification/src/Notification.tsx

```diff
@@ -81,7 +81,7 @@ export const Notification = compose<NotificationType>({
 
     return (final: NotificationProps, ...children: React.ReactNode[]) => {
       const { variant, icon, title, action, onActionPress, ...rest } = mergeProps(userProps, final);
-      const mergedProps = mergeProps<PressableProps>(rest, rootStyle);
+      const mergedProps = mergeProps<PressableProps>(rest, { style: rootStyle });
```

Comment:
Change rootStyle to return {style: /*whatever the internals are*/ instead of creating the new style object here.

Reply:
The test should tell you why it failed, but if it's not you can change compareTrees in enzymeTests.ts to print out log statements instead of throwing. Beyond that it's figuring out why something changed when it shouldn't have. General guidelines for preventing wasted renders can be found on sites like https://www.freecodecamp.org/news/how-to-identify-and-resolve-wasted-renders-in-react-cc4b1e910d10/

Common culprits are objects (which include arrays and functions) that are recreated on every render - React only does a shallow compare to figure out if props have changed, so new objects get counted as a changed prop even if their internals are the same.

Reply:
Tried that, it complained about not having any properties in common with ViewStyle
<img width="713" alt="Screen Shot 2022-09-16 at 12 04 32 PM" src="https://user-images.githubusercontent.com/78454019/190713169-edb9b318-e8f6-4115-91ef-9a920ee28ae4.png">

So I removed the type of rootStyle, and then the mergeProps line started complaining:
<img width="863" alt="Screen Shot 2022-09-16 at 12 05 58 PM" src="https://user-images.githubusercontent.com/78454019/190712998-482de3dc-9c4e-431e-86b8-fabad887c750.png">

And then this is something else I tried, skipping mergeProps entirely but the render tests are still failing with this:
<img width="614" alt="Screen Shot 2022-09-16 at 12 02 29 PM" src="https://user-images.githubusercontent.com/78454019/190713256-7ba95d6c-a952-48cb-bd28-6f287755b7ed.png">

Also another thing I tried with rootStyleObject
<img width="856" alt="Screen Shot 2022-09-16 at 12 03 49 PM" src="https://user-images.githubusercontent.com/78454019/190713242-96d9fac8-21b2-4685-8d48-0652823da874.png">

Reply:
The return type changes once you change it to return {style: {styleInternals}} as opposed to {styleInternals}

Reply:
Specifically you could try setting the return type to ViewProps['style'] or NotificationProps

Reply:
Okay updating the type to just ViewProps got rid of all the red errors, but the render tests still fail with this

<img width="621" alt="Screen Shot 2022-09-16 at 12 43 37 PM" src="https://user-images.githubusercontent.com/78454019/190718196-453c1ead-bd8c-4014-9da7-aff8a3ed6d41.png">

Reply:
Where does the test say it's failing?

Reply:
<img width="704" alt="Screen Shot 2022-09-16 at 1 12 07 PM" src="https://user-images.githubusercontent.com/78454019/190725156-7559d7d8-4a89-4fbe-9536-290f8e4fe322.png">

Also committed/pushed the code above so the pipelines should show the error when they're done running

Reply:
Okay I misunderstood which test was failing, I thought it was the rerender test (i.e. force an update and make sure the component stays the same) and not the render consistency test (if you render two copies of the same component, the props are all the same since we memoize stuff).

It might be the case that in this component's case, this test is expected to fail. Notification creates a new style. I'll have to think about it a bit

Reply:
Okay so there is a way to get it to work. You need to store the rootStyle on the larger style cache so that the style can be reused across instances of Notification. The way you do that is to create a function which takes in the inputs for creating the rootStyle and returns the root style, and then use memoize from our framework to wrap it (which uses a framework-wide cache) instead of useMemo, which is per-component.

I don't understand why it's not failing already though.. I would kind of expect it to fail as it is the way the component is set up.

Reply:
No sorry my fault for the confusion, I didn't really distinguish between the rerender test/render consistency test in my head because they've usually failed together, I think I wrote render consistency when it should have been rerender

Will try your suggestion. That kind of seems like how getShadowTokenStyleSet works?

Reply:
That worked locally! Although now I don't really understand only rootStyle needed memoize and not messageStyle, seems like the messageStyle has the same issue rootStyle used to have and should cause the test to fail

Reply:
We limit how far the test goes into the render tree. I imagine that if we increase the depth to 5 or something, it would start failing? For completeness it would make sense to fix it for the text styling, too.

Reply:
Oh right yeah I think I noticed that at some point with this PR, when the Shadow tests were failing it was only complaining about the outer view when the inner view had the same issue. Will update messageStyle